src/app-layer-htp.c

```diff
@@ -4264,82 +3983,53 @@ libhtp:\n\
 
     HTPConfigure();
 
-    if (cfglist.cfg == NULL) {
-        printf("No default config created.\n");
-        goto end;
-    }
+    FAIL_IF_NULL(cfglist.cfg);
 
-    if (cfgtree == NULL) {
-        printf("No config tree created.\n");
-        goto end;
-    }
+    FAIL_IF_NULL(cfgtree);
 
     htp_cfg_t *htp = cfglist.cfg;
     uint8_t buf[128];
     const char *addr;
     void *user_data = NULL;
 
     addr = "192.168.10.42";
-    if (inet_pton(AF_INET, addr, buf) == 1) {
-        (void)SCRadixFindKeyIPV4BestMatch(buf, cfgtree, &user_data);
-        if (user_data != NULL) {
-            HTPCfgRec *htp_cfg_rec = user_data;
-            htp = htp_cfg_rec->cfg;
-            SCLogDebug("LIBHTP using config: %p", htp);
-        }
-        if (htp == NULL) {
-            printf("Could not get config for: %s\n", addr);
-            goto end;
-        }
-    }
-    else {
-        printf("Failed to parse address: %s\n", addr);
-        goto end;
-    }
+    FAIL_IF(inet_pton(AF_INET, addr, buf) != 1);
+    (void)SCRadixFindKeyIPV4BestMatch(buf, cfgtree, &user_data);
+    FAIL_IF_NULL(user_data);
+    HTPCfgRec *htp_cfg_rec = user_data;
+    FAIL_IF_NULL(htp);
+    htp = htp_cfg_rec->cfg;
```

Comment:
Juding from the original test, it seems to make more sense to invert these two lines, that is, have the check for `htp == NULL` _after_ we've assigned the new value to it.
(This one is on me, I see that I suggest it just as you did. Sorry about that)